### PR TITLE
Gather method

### PR DESCRIPTION
**Pull request by davidmeunier79**

### CODE REVIEW

**Comment by davidmeunier79** [comment]
Is it ok to go? What is it with the codecov? Sorry I did not follow the last development on ephypype

**Comment by jasmainak** [comment]
@davidmeunier79 you have to make Travis and Circle happy. That is why they were added :)

File: ephypype/tests/test_spectral.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    try:
+        res_plv = compute_spectral_connectivity(
+            data=ts_mat_trials,
+            con_method="plv",
+            mode="multitaper",
+            fmin=fmin,
+            fmax=fmax,
+            sfreq=sfreq,
+            gathering_method='mean')
+    except SystemExit:
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: this is not yet fixed ...

File: ephypype/tests/test_spectral.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    try:
+        res_plv = compute_spectral_connectivity(
+            data=ts_mat,
+            con_method="plv",
+            mode="multitaper",
+            fmin=fmin,
+            fmax=fmax,
+            sfreq=sfreq,
+            gathering_method='mean')
+    except SystemExit:
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: There should be no `try-except` statements in tests ...

File: ephypype/tests/test_spectral.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        all_data=ts_mat_trials,
+        con_method="coh",
+        mode="multitaper",
+        fmin=fmin,
+        fmax=fmax,
+        sfreq=sfreq,
+        gathering_method='mean',
+        save_dir="/tmp/ephypype_test")
+
+if __name__ == '__main__':
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: you should remove this and use `pytest` to run the tests.

File: ephypype/tests/test_spectral.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                               compute_and_save_multi_spectral_connectivity)  # noqa
+
+import ephypype
+
+print(ephypype.__path__)
+
+print(ephypype.__version__)
+
+#data_path = mne.datasets.sample.data_path()
+#raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: there should be no code comments ... please remove them if you don't need them

File: ephypype/tests/test_spectral.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            mode="multitaper",
+            fmin=fmin,
+            fmax=fmax,
+            sfreq=sfreq,
+            gathering_method='mean')
+    except SystemExit:
+        print("SystemExit should not raise excepction for plv with several trials")
+        assert False
+
+    #assert np.all(res_mean <= res_max), "error, all mean values should be lower than max values"
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: same comment here. No code comments.

**Comment by davidmeunier79** [comment]
> @davidmeunier79 you have to make Travis and Circle happy. That is why they were added :)

I was refering to the 2 codedev, the ones that gave me failed checks. I can see Travis Appveyor and Circle were working

**Comment by jasmainak** [comment]
Please don't merge your own PRs. It's not considered good practice ...

also the merge commits were not removed yet. It may seem like a nitpick, but in the long run, it can make it really hard to track things and find what introduced a bug.

**Comment by jasmainak** [comment]
@davidmeunier79 I undid your merge. Can you make a new PR with clean commits?

If you need somewhere to start, I made a backup of your branch [here](https://github.com/jasmainak/ephypype/tree/gather_method)

**Comment by davidmeunier79** [comment]
Sorry but it took me a long time to this merge, it was a real pain to try to contribute to ephypye with my previous (and I admit not very clean)  way of doing. And I have no clue what you are refering to ("merge commits" + you undid my merge, I see the branch is the same state as it was yesterday. Are you referring to the multiple commit messages?

**Comment by davidmeunier79** [comment]
> @davidmeunier79 I undid your merge. Can you make a new PR with clean commits?
> 
> If you need somewhere to start, I made a backup of your branch [here](https://github.com/jasmainak/ephypype/tree/gather_method)

I have done a new PR for gather_method only starting from your backup https://github.com/neuropycon/ephypype/pull/44

**Comment by jasmainak** [comment]
@davidmeunier79 this is a merge commit:

merge_commit

It typically happens when you do `git merge` or `git pull`. You should almost never use those commands in the first place. For professional software development, only use `git rebase` command. The reason is that you want to do only "fast-forward" merges. That is to say, if master branch looks like this:

o-o-o-o-x-x-x (master)

where 'o' and 'x' are commits, your branch should always look like this:

o-o-o-o-x-x-x-y-y-y (feature_branch)

and not:

o-y-o-o-o-y-x-x-x-y (feature_branch)

where 'y' are the commits from your pull request. The commit history should be in blocks of "pull requests". The only way to ensure this is by doing a rebase, not a merge.

By undoing your merge, I meant that I removed your commits from `master` branch and force pushed. Ideally, we should not be doing a force push, but I did it so we can keep a clean commit history. 

Sorry, git is a bit of a pain in the beginning, but once you get a hang of it, it's easy.

**Comment by davidmeunier79** [comment]
hopefully solved by https://github.com/neuropycon/ephypype/pull/52

**Pull request merged**